src/tools/rbd/action/Kernel.cc

```diff
@@ -188,6 +221,10 @@ static int do_kernel_map(const char *poolname, const char *imgname,
 
   free(devnode);
 out:
+  if (r < 0) {
+    rbd_check_unsup_features(poolname, imgname, snapname, errno);
```

Comment:
errno could have changed since the write(2) call - just use r instead, since it's already been copied there in safe_write()